remediation_worker/jobs/kinesis_encrypt_stream/kinesis_encrypt_stream.py

```diff
@@ -0,0 +1,173 @@
+            client: Instance of the AWS boto3 client.
+        """
+
+        logging.info(
+            "Beginning kinesis data stream encryption evaluation."
+        )
+
+        # Check to see if kinesis data stream is encrypted
+        encryption_type = self.check_encryption(client, stream_name)
+
+        # If kinesis data stream is not encrypted
+        if encryption_type == "NONE":
+            self.enable_encryption(client, stream_name)
+        # If encryption is enabled
+        # We have nothing to do.
+        # log and exit
+        elif encryption_type == "KMS":
+            logging.info(
+                "Kinesis data stream %s is already encrypted, taking no action.",
+                stream_name
+            )
+            sys.exit(0)
+        # If anything else returns,
+        # log and exit
+        else:
+            logging.error(
+                "An unknown encryption type returned for kinesis data stream %s. Exiting.",
+                stream_name
+            )
+            sys.exit(1)
+
+    def check_encryption(self, client, stream_name):
+        """Checks the kinesis data stream encryption setting.
+        Args:
+            client: Instance of the AWS boto3 client
+            stream_name ([str]): The name of the kinesis stream.
+        Returns:
+            [string]: "NONE" if no encryption is enabled.
+        """
+
+        # Check if stream is encrypted
+        try:
+            encryption_type = client.describe_stream(
+                StreamName=stream_name)["StreamDescription"]["EncryptionType"]
+        except client.exceptions.ResourceNotFoundException as found_error:
+            logging.error(
+                "A failure occured with error: %s. Check if the resource exists.",
+                format(found_error.response['Error']['Code'])
+            )
+            sys.exit(1)
+
+        return encryption_type
+
+    def enable_encryption(self, client, stream_name):
+        """Enable encryption on the kinesis data stream.
+        Args:
+            client: Instance of the AWS boto3 client.
+            stream_name ([str]): The name of the kinesis stream.
+        """
+        logging.info(
+            "Enabling encryption on kinesis data stream %s",
+            stream_name
+        )
+        try:
+            client.start_stream_encryption(
+                StreamName=stream_name,
+                EncryptionType="KMS",
+                KeyId="alias/aws/kinesis"  # KEK owned by Kinesis Data Streams
+            )
+            logging.info(
+                "Encryption enabled on kinesis data stream %s",
+                stream_name
+            )
+        except client.exceptions.ResourceInUseException as use_error:
+            if use_error.response["Error"]["Code"] == "ResourceInUseException":
+                logging.info(
+                    "Kinesis data stream %s is in an unavailable state.",
+                    stream_name
+                )
+            sys.exit(1)
+        except:
```

Comment:
Can you please log the exact error returned by AWS instead of giving the generic error. It will be more helpful and informative.

Reply:
Changed the exception to ClientError and log all AWS error accordingly.